src/Symfony/Bundle/FrameworkBundle/Secret/EncoderInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Secret;
+
+/**
+ * EncoderInterface defines an interface to encrypt and decrypt messages.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+interface EncoderInterface
+{
+    /**
+     * Generate the keys and material necessary for it operation.
+     *
+     * @param bool $override Override previous keys if already exists
+     *
+     * @return string[] List of resources created
+     */
+    public function generateKeys(bool $override = false): array;
+
+    /**
+     * Encrypt a message.
+     */
+    public function encrypt(string $message): string;
+
+    /**
+     * Decrypt a message.
+     */
+    public function decrypt(string $encryptedText): string;
```

Comment:
`$encryptedText` -> `$encryptedMessage` for consistency with PHPdoc and `encrypt($message)` ?

Reply:
renamed:
- `text` and `messages` into `secret`
- `plain text` into `decrypted` or `secret`